api/api_controller.go

```diff
@@ -265,7 +265,12 @@ func (c *Controller) SetupLakeFSHandler() setupop.SetupLakeFSHandler {
 		c.deps.Collector.CollectEvent("global", "init")
 
 		username := swag.StringValue(setupReq.User.Username)
-		cred, err := auth.CreateInitialAdminUser(c.deps.Auth, c.deps.MetadataManager, username)
+		var cred *model.Credential
+		if setupReq.User.Key == nil {
+			cred, err = auth.CreateInitialAdminUser(c.deps.Auth, c.deps.MetadataManager, username)
+		} else {
```

Comment:
I hate swagger and its defaults for required/optional, but I remember there's a way you'll get empty strings here, right?
If so, need to verify here.